crates/supervisor/storage/src/models/log.rs

```diff
@@ -32,20 +111,93 @@ pub struct LogEntry {
 /// - `log_index` (`u64`): The index of the log entry within the block where the message was logged.
 /// - `timestamp` (`u64`): The timestamp associated with the block where the message was recorded.
 /// - `hash` (`B256`): The unique hash identifier of the message.
-#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize, Deserialize, Compact)]
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize, Deserialize)]
 pub struct ExecutingMessageEntry {
+    /// Log index within the block.
+    pub log_index: u32,
     /// ID of the chain where the message was emitted.
     pub chain_id: u64,
     /// Block number in the source chain.
     pub block_number: u64,
-    /// Log index within the block.
-    pub log_index: u64,
     /// Timestamp of the block.
     pub timestamp: u64,
     /// Hash of the message.
     pub hash: B256,
 }
 
+/// Compact encoding for [`ExecutingMessageEntry`] used in log storage.
+///
+/// This format ensures deterministic encoding and lexicographic ordering by
+/// placing `log_index` first, which is used as the subkey in dup-sorted tables.
+///
+/// ## Encoding Layout (ordered):
+/// - `log_index: u32` – Subkey for dup sort ordering.
```

Comment:
I think there is a typo. This isn't the subKey. Right?
This is the initiating message identifier. SubKey would be `LogEntry.index`.